app/src/main/java/com/quran/labs/androidquran/view/AudioStatusBar.java

```diff
@@ -409,6 +453,83 @@ private void addDownloadProgress(@StringRes int text) {
     addView(ll, lp);
   }
 
+  @SuppressLint("NewApi")
+  private void showRecitationListeningMode() {
+    currentMode = RECITATION_LISTENING_MODE;
+    removeAllViews();
+
+    ImageView recitationButton = new ImageView(context);
+    recitationButton.setImageTintList(ColorStateList.valueOf(Color.CYAN));
+
+    if (isRtl) {
+      addButton(recitationButton, R.drawable.ic_mic, false);
+      addSeparator();
+      addButton(R.drawable.ic_transcript, false);
+      addSeparator();
+      addSpacer();
+      addSeparator();
+      addButton(R.drawable.ic_hide_page, false);
+    } else {
+      addButton(R.drawable.ic_hide_page, false);
+      addSeparator();
+      addSpacer();
+      addSeparator();
+      addButton(R.drawable.ic_transcript, false);
+      addSeparator();
+      addButton(recitationButton, R.drawable.ic_mic, false);
+    }
+  }
+
+  private void showRecitationStoppedMode() {
```

Comment:
what do you think about swapping this out for a compose audio bar and toggle buttons based on state instead (or xml with show/hide calls? this was already a bit unwieldy and becomes troublesome if we keep trying to expand it like this

Reply:
Yes, that would be better. This is a much bigger change however, so may be better as a separate PR